Assets/MixedRealityToolkit.SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2232 @@
+
+        // Hand position when starting a motion
+        private Vector3 initialPointerPos;
+
+        // Hand position previous frame
+        private Vector3 lastPointerPos;
+
+        [SerializeField]
+        [HideInInspector]
+        /// <summary>
+        /// The distance in front of the scroller, in local space, for a touch release.
+        /// We serialize and then hide this because we want to use the value
+        /// when drawing the touch plane, but not actually expose the value publicly.
+        /// </summary>
+        private float releaseDistance;
+
+        // Current time at initial press
+        private float initialPressTime;
+
+        #endregion drag position calculation variables
+
+        #region velocity calculation variables
+
+        // Simple velocity of the scroller: current - last / timeDelta
+        private float scrollVelocity = 0.0f;
+
+        // Filtered weight of scroll velocity
+        private float avgVelocity = 0.0f;
+
+        // How much we should filter the velocity - yes this is a magic number. Its been tuned so lets leave it.
+        private readonly float velocityFilterWeight = 0.97f;
+
+        // Simple state enum to handle velocity logic
+        private enum VelocityState
+        {
+            None = 0,
+            Resolving,
+            Calculating,
+            Bouncing
+        }
+
+        // Internal enum for tracking the velocity state of the list
+        private VelocityState velocityState = VelocityState.None;
+
+        // Pre calculated destination with velocity and falloff when using per item snapping
+        private Vector3 velocityDestinationPos;
+
+        // Velocity container for storing previous filtered velocity
+        private float velocitySnapshot;
+
+        #endregion velocity calculation variables
+
+        // The Animation CoRoutine
+        private IEnumerator animateScroller;
+
+        #region ObjectCollection methods
+
+        /// <inheritdoc/>
+        public override void UpdateCollection()
+        {
+            // Generate our scroll specific objects
+
+            // ScrollContainer empty game object null check - ensure its set up properly
+            if (scrollContainer == null)
+            {
+                Transform oldContainer = transform.Find("Container");
+
+                if (oldContainer != null)
+                {
+                    scrollContainer = oldContainer.gameObject;
+                    Debug.LogWarning(name + " ScrollingObjectCollection found an existing Container object, using it for the list");
+                }
+                else
+                {
+                    scrollContainer = new GameObject();
+                }
+
+                scrollContainer.name = "Container";
+                scrollContainer.transform.parent = transform;
+                scrollContainer.transform.localPosition = Vector3.zero;
+                scrollContainer.transform.localRotation = Quaternion.identity;
+            }
```

Comment:
Create sub function => SetupScrollContainer or UpdateScrollContainer?

Same with ClippingObject, clip box and other sections below?